integrations/terraform-mwi/provider/kubernetes_data_source.go

```diff
@@ -78,18 +123,39 @@ func (d *KubernetesDataSource) Configure(
 	// TODO: Fetch bot config data from the provider.
 }
 
-func (d *KubernetesDataSource) Read(
+func (r *KubernetesDataSource) loadModelAndSetDefaults(
 	ctx context.Context,
 	req datasource.ReadRequest,
 	resp *datasource.ReadResponse,
-) {
+) KubernetesDataSourceModel {
 	var data KubernetesDataSourceModel
 	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return data
+	}
+
+	// Set default for credential TTL if not provided.
+	if data.CredentialTTL.IsNull() || data.CredentialTTL.IsUnknown() {
+		data.CredentialTTL = timetypes.NewGoDurationValue(
+			time.Minute * 30,
+		)
+	}
```

Comment:
Ah, guessing `datasource/schema` doesn't support the `Default` field like `resource/schema` does?